workspaces/ui-v2/src/pages/diffs/contexts/SimulatedCommandContext.tsx

```diff
@@ -22,6 +26,7 @@ export const SimulatedCommandContext = React.createContext<SimulatedCommandConte
 export function SimulatedCommandStore(props: SimulatedCommandStoreProps) {
   const value = { previewCommands: props.previewCommands };
   const [isProcessing, setIsProcessing] = useState(true);
+  const store = useMemo(() => createReduxStore(), []);
```

Comment:
Wait, so we create an entire new app state for simulated commands? I'm guessing the assumption here is that all the state in the main store is invalidated by a spec change? It _seems_ like that's a bit much and we'd want something more surgical, but at the same time I can't tell you what that would be. Could also happen later, of course.

Reply:
So the simulated command store is used for us to "preview" the state of say an endpoint after a diff is applied and learnt. We need a spectacle instance to run in memory, and likewise need a completely different state to represent the "simulated commands" - since the source of truth is not the main redux store (since we haven't applied the commands). Now I agree this is fairly heavy weight to load the redux store - and the things we need to load up are a subset of this (endpoints, shapes, anything in the EndpointDocumentationPanel)

The reasons I decided to pull in the new redux store is:
- It allows us to use the same appSelectors regardless of where it is (we can reuse our existing components that already implicitly rely on this behavior)

- A new redux store is effectively a large object with no initial data (we control the initial fetch state) - which _shouldn't_ be too much overhead to load in

The alternative is to either make any component that uses the simulated command store "render only" and accept props for their data, build a context wrapper that handles the simulated command store or to figure out how not load the entire redux store

Creating a new store and keeping the same interface is a tradeoff of iteration speed and simplicity against performance (i'm unclear on what / how much performance gains there would be here). It feels a little odd, but I've come to think this makes the most sense in the simulated command context - i.e. each simulated command context has it's own "state" (think of it as an iframe)